drivers/hwinfo/hwinfo_silabs_series2.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2025 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/hwinfo.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <string.h>
+
+#include <em_system.h>
+#include <em_rmu.h>
+
+#define IF_EXISTS(x) COND_CODE_1(defined(x), (x), (0))
```

Comment:
Not related: It is the second time in one week [we need this pattern][1]. Maybe we could generalize it.

[1]: https://github.com/zephyrproject-rtos/zephyr/pull/83442/commits/762c3d2beb7aee6000a85747cc24a134d376c485#r1910593240

Reply:
I was thinking the same thing, i.e. that a generic macro in `include/zephyr/sys/util_macro.h` would make a lot of sense. I can think of a lot of places in Bluetooth code that could take advantage of it. There's already something similar in the form of `UTIL_OR()`, however that doesn't support passing a potentially undefined macro, so something new is needed. I'd still consider having `*_OR()` as the name to be consistent with existing similar macros (especially those for DT), and there could be a convenience variant for the use case in this PR called `*_OR_ZERO()` if we like.